core/src/components/item-sliding/test/basic/item-sliding.e2e.ts

```diff
@@ -68,6 +68,28 @@ configs({ modes: ['md'], directions: ['ltr'] }).forEach(({ title, screenshot, co
       await expect(item).toHaveScreenshot(screenshot(`item-sliding-gesture`));
     });
 
+    test('the dynamic element should be clicked', async ({ page }) => {
+      await page.goto(`/src/components/item-sliding/test/basic`, config);
+      const item = page.locator('#item43');
+
+      // Scroll page in view and drag the item-sliding
+      await page.setIonViewport();
+      await dragElementBy(item, page, -150);
+
+      // Trigger the button to add the dynamic content
+      const button = page.locator('#add-element-btn');
+      await button.click();
+
+      // Check if the element added is clicked
+      const elementAdded = page.locator('#element-added-delete');
+      await elementAdded.click();
+
+      await page.waitForChanges();
+
+      // item-sliding doesn't have an easy way to tell whether it's fully open so just screenshot it
+      await expect(item).toHaveScreenshot(screenshot(`dynamic-element-clicked`));
```

Comment:
I think we may be able to avoid a screenshot test here. 

The behavior we want to validate with this PR is if the newly inserted button is focusable. If it cannot receive focus after trying to click the locator, the test will fail.

Something similar to:
```ts
test('the dynamic element should be clicked', async ({ page }) => {
  await page.setContent(`
    /* We can move the specific behavior we are testing here 
      to avoid visual changes to the existing screenshots captured for 
      src/components/item-sliding/test/basic.
      The content here will be evaluated as HTML. Wrap JS in a <script> */
  `, config);

  // Drag the element
  // Click the add button
  // Find the added button
  // Click the button
  // Check if the added button has focused
});
```

The Playwright assertion that can be used here is [`toBeFocused`](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-focused). 

Example of an E2E test using `.setContent`: https://github.com/ionic-team/ionic-framework/blob/bf7922c8b37b32dbf90650cb74a2e77bedf0c118/core/src/components/item/test/basic/item.e2e.ts#L18-L39

Reply:
Thanks for the suggestion. Already applied it and commited.